woocommerce/payment-gateway/External_Checkout/Admin.php

```diff
@@ -288,6 +291,192 @@ protected function get_display_location_options() {
+
+		$this->add_configuration_warnings_notices();
+	}
+
+
+	/**
+	 * Adds one error notice for all configuration options that need attention.
+	 *
+	 * @since 5.10.0
+	 */
+	protected function add_configuration_errors_notice() {
+
+		$errors = $this->get_configuration_errors();
+
+		if ( ! empty( $errors ) ) {
+
+			$message = sprintf(
+				/* translators: Placeholders:  - external checkout label, %2$s - <strong> tag, %3$s - </strong> tag */
+				__( '%2$S%1$s is disabled.%3$S', 'woocommerce-plugin-framework' ),
+				$this->handler->get_label(),
+				'<strong>', '</strong>',
+			);
+
+			if ( 1 === count( $errors ) ) {
+				$message .= ' ' . current( $errors );
+			} else {
+				$message .= '<ul><li>' . implode( '</li><li>', $errors ) . '</li></ul>';
+			}
+
+			$this->handler->get_plugin()->get_admin_notice_handler()->add_admin_notice( $message, $this->section_id . '-configuration-issue', [
+				'notice_class' => 'error',
+				'dismissible'  => false,
+			] );
+		}
+	}
+
+
+	/**
+	 * Adds warning notices for each configuration option that may need attention.
+	 *
+	 * @since 5.10.0
+	 */
+	protected function add_configuration_warnings_notices() {
+
+		if ( $this->should_display_shipping_based_tax_notice() ) {
+
+			$message = $this->get_shipping_based_tax_notice();
+
+			$this->handler->get_plugin()->get_admin_notice_handler()->add_admin_notice( $message, $this->section_id . '-shipping-based-tax', [
+				'notice_class' => 'warning',
+				'dismissible'  => true,
+			] );
+		}
+
+		if ( $this->should_display_billing_based_tax_notice() ) {
+			$message = $this->get_billing_based_tax_notice();
+
+			$this->handler->get_plugin()->get_admin_notice_handler()->add_admin_notice( $message, $this->section_id . '-billing-based-tax', [
+				'notice_class' => 'warning',
+				'dismissible'  => true,
+			] );
+		}
+	}
+
+
+	/**
+	 * Gets the error messages for configuration issues that need attention.
+	 *
+	 * @since 5.10.0
+	 *
+	 * @return string[] error messages
+	 */
+	protected function get_configuration_errors() {
+
+		$errors = [];
+
+		// currency notice
+		$accepted_currencies = $this->handler->get_accepted_currencies();
+
+		if ( ! empty( $accepted_currencies ) && ! in_array( get_woocommerce_currency(), $accepted_currencies, true ) ) {
+
+			$errors[] = sprintf(
+				/* translators: Placeholders: %1$s - plugin name, %2$s - a currency/comma-separated list of currencies, %3$s - <a> tag, %4$s - </a> tag, %5$s - external checkout label */
+				_n(
+					'Accepts payment in %1$s only. %2$sConfigure%3$s WooCommerce to accept %1$s to enable %4$s.',
+					'Accepts payment in one of %1$s only. %2$sConfigure%3$s WooCommerce to accept one of %1$s to enable %4$s.',
+					count( $accepted_currencies ),
+					'woocommerce-plugin-framework'
+				),
+				'<strong>' . implode( ', ', $accepted_currencies ) . '</strong>',
+				'<a href="' . esc_url( admin_url( 'admin.php?page=wc-settings&tab=general' ) ) . '">',
+				'</a>',
+				$this->handler->get_label(),
```

Comment:
Trailing comma fatal
```suggestion
				$this->handler->get_label()
```